File: src/Contexts/Rutine/Exercises/domain/ExerciseId.ts
import { InvalidArgumentError } from '../../../Shared/domain/value-object/InvalidArgumentError';
import { StringValueObject } from '../../../Shared/domain/value-object/StringValueObject';

export class ExerciseId extends StringValueObject {
  constructor(value: string) {
    super(value.toUpperCase());
    this.ensureLengthIsLessThan12Characters(value);
  }

  private ensureLengthIsLessThan12Characters(value: string): void {
    if (value.length > 12) {
      throw new InvalidArgumentError(`The Exercise Name <${value}> has more than 12 characters`);
    }
  }
}

File: src/main/java/com/kuro/rutineweb/Shared/entities/ExerciseId.java
package com.kuro.rutineweb.Shared.entities;

import com.kuro.rutineweb.Shared.entities.valueObject.InvalidArgumentError;
import com.kuro.rutineweb.Shared.entities.valueObject.StringValueObject;

public class ExerciseId extends StringValueObject {
    public ExerciseId(String value) {
        super(value.toUpperCase());
        ensureLengthIsLessThan12Characters(value);
    }

    private void ensureLengthIsLessThan12Characters(String value) {
        if(value.length() > 12) {
            throw new InvalidArgumentError("The Exercise Id " + value + " has more than 12 characters");
        }
    }
}
